OSDOCS-3774: OLM 4.11 Known issue PSA warning on catalog sources

michaelryanpeter opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OSDOCS-<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch. --->

Version(s): 4.11
<!--- Specify the version or versions of OpenShift your PR applies to.

Examples:
  * PR applies to all versions after a specific version (e.g. 4.8): 4.8+
  * PR applies to the in-development version (e.g. 4.11) and future versions: 4.11+
  * PR applies only to a specific single version (e.g. 4.10): 4.10
  * PR applies to multiple specific versions (e.g. 4.6-4.8): 4.6, 4.7, 4.8 --->

Issue: [OSDOCS-3774](https://issues.redhat.com//browse/OSDOCS-3774)
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview (Requires VPN):
[OCP 4.11 Known issues (scroll down to find the known issue)](http://file.rdu.redhat.com/mipeter/osdocs-3774-olm-4.11-ki-psa-catalog/release_notes/ocp-4-11-release-notes.html#ocp-4-11-known-issues)
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build.

NOTE:
Automatic preview functionality is currently not available.
  * OpenShift documentation team members (core and aligned) must include a link to a locally generated preview for PRs that update the rendered build in any way.
  * External contributors can request a generated preview from the OpenShift documentation team. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->



<!--- Next steps after opening your PR:

* Ask for peer review from the OpenShift docs team:
  - For Red Hat associates: Ping @peer-review-squad requesting a review in the #forum-docs-review channel (CoreOS Slack workspace) and provide the following information:
    * A link to the PR.
    * The size of the PR that the GitHub bot assigns (ex: XS, S, M, L, XL).
    * If there is urgency or a deadline for the review.
  - For community authors: Request a review by tagging @openshift/team-documentation in a GitHub comment.

  Slack is the quickest and preferred way to request a review.

* IMPORTANT:
  - All documentation changes must be verified by a QE team associate before merging.
  - Squash to one commit before submitting your PR for peer review.

* For more information about verifying your content, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content

* For more information about contributing to OpenShift documentation, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/contributing.adoc

Additional resources

The OpenShift docs repo adheres to the following style guides:

- OpenShift documentation guidelines (OSDOCS)
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc
- Red Hat Supplementary Style Guide (SSG)
  https://redhat-documentation.github.io/supplementary-style-guide/
- Modular Documentation Reference Guide (Mod Docs)
  https://redhat-documentation.github.io/modular-docs/
- IBM Style Guide (ISG)
  https://www.ibm.com/docs/en/ibm-style

You can log in to the ISG by using your @redhat.com id and single sign-on (SSO) credentials. --->

perdasilva commented:
Perfection! Many thanks, Mike!!!

grokspawn commented:
/chefs-kiss

michaelryanpeter commented:
@Xia-Zhao-rh: Would you PTAL at this known issue for the 4.11 release notes?

Xia-Zhao-rh commented:
cc @jianzhangbjz

jianzhangbjz commented on release_notes/ocp-4-11-release-notes.adoc:
```diff
@@ -1055,6 +1055,25 @@ This script removes unauthenticated subjects from the following cluster role bin
++
+.Example `manifests/<namespace>.yaml` file
++
+[source,yaml]
```
Suggest adding the `resource` name here so that the user clearly know add this `label` to `namespace` resource. Such as
```yaml
apiVersion: v1
kind: Namespace
metadata:
...
```

michaelryanpeter replied:
Addressed

jianzhangbjz commented on release_notes/ocp-4-11-release-notes.adoc:
```diff
@@ -1055,6 +1055,25 @@ This script removes unauthenticated subjects from the following cluster role bin
 
+* Currently, Operator Lifecycle Manager (OLM) displays pod security admission (PSA) warnings for catalog sources installed outside of the `openshift-marketplace` namespace. 
++
+Workaround: Cluster administrators can remove the warning by setting the level of the audit mode to `baseline`, as shown in the following example:
```
It's better to add the `warn` mode too, such as,
```yaml
    pod-security.kubernetes.io/audit: baseline
    pod-security.kubernetes.io/warn: baseline
```

Xia-Zhao-rh commented:
/lgtm

jianzhangbjz commented:
/hold

Xia-Zhao-rh commented:
/remove-lgtm

michaelryanpeter commented:
@Xia-Zhao-rh and @jianzhangbjz: Thank you for the feedback. 

I addressed your comments and made a few other changes. Would you please take another look?

xingxingxia commented:
CC @y4sht

jianzhangbjz commented:
Thanks! 
/unhold
/lgtm

anik120 commented on release_notes/ocp-4-11-release-notes.adoc:
```diff
@@ -1055,6 +1055,29 @@ This script removes unauthenticated subjects from the following cluster role bin
+<1> Set the `audit` mode label to `baseline`.
+<2> Set the `warn` mode label to `baseline`.
++
+For more information, see link:https://kubernetes.io/docs/concepts/security/pod-security-admission/#pod-security-admission-labels-for-namespaces[Pod Security Admission labels for namespaces]. (link: https://bugzilla.redhat.com/show_bug.cgi?id=2100323[*BZ#2100323*])
```
/hold

There is some confusion surrounding the bz. Holding this till that gets resolved.

anik120 commented:
With Pod Security Admission profile `restricted` enabled by default for all namespaces (see [1] and [2]), installing a `CatalogSource` built with a version of opm lesser than v1.23.0 throws an error `Error: open ./db-xxxx: permission denied`. 

Workaround: Cluster administrators allow the creation of such a CatalogSource by setting the required PSA labels in the namespace metadata. Apply the `privileged` label to the `warn` and `audit` modes, as shown in the following example:

.Example `manifests/<target_namespace>.yaml` file
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
...
  labels:
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/audit: privileged <1>
    pod-security.kubernetes.io/warn: privileged <2>
  name: <namespace-name>
----
<1> Set the `audit` mode label to `privileged`.
<2> Set the `warn` mode label to `privileged`.

For more information, see  https://bugzilla.redhat.com/show_bug.cgi?id=2100323[*BZ#2100323*])

[1] http://file.rdu.redhat.com/~mbridges/11JUL2022/psa/authentication/understanding-and-managing-pod-security-admission.html
[2] http://file.rdu.redhat.com/~mbridges/auth-133-rn/release_notes/ocp-4-11-release-notes.html#ocp-4-11-auth-pod-security-admission

@michaelryanpeter @perdasilva this doc update should probably be laid out like this instead ☝🏽

michaelryanpeter commented:
I am closing this PR because I messed up my commit on a rebase. I will open a new PR and reference this PR in a comment.
